src/include/ipc/dai-intel.h

```diff
@@ -99,9 +99,11 @@ struct sof_ipc_dai_hda_params {
 struct sof_ipc_dai_alh_params {
 	uint32_t reserved0;
 	uint32_t stream_id;
+	uint32_t rate;
+	uint32_t channels;
```

Comment:
We have two sources of configurations for channels:
- a static configuration of the demux, with channel information coming from topology
- a dynamic configuration of ALH, with channel information coming from ALSA core which doesn't know how the demux was configured.

It's likely we will see a mismatch, or you are assuming that the demux always provides the same number of channels that was provided in input, in which case it's used as a Tee. If there is a selection, then this probably breaks, no?

Reply:
It would also be useful to clarify what happens on capture? what does channels represent?

Reply:
We assume the demux configuration in the topology is going to be correct given the expected format on the DAIs.

Capture is the same as with playback, it's the format programmed on the DAI. So, if the capture stream coming from a DAI has 2 channels, this value will be set to 2.